nengo/tests/test_connection.py

```diff
@@ -611,6 +612,23 @@ def test_set_eval_points(Simulator):
     Simulator(model)  # Builds fine
 
 
+@pytest.mark.parametrize('eval_points, radius', [
+    (UniformHypersphere().sample(500, 1), 0.5),
+    (UniformHypersphere().sample(500, 1), 1.),
+    (UniformHypersphere().sample(500, 1), 1.5)])
+def test_eval_points_scaling(Simulator, eval_points, radius, seed):
+    model = nengo.Network(seed=seed)
+    with model:
+        a = nengo.Ensemble(1, 1, radius=radius)
+        b = nengo.Ensemble(1, 1)
+        con = nengo.Connection(a, b, eval_points=eval_points)
+
+    sim = Simulator(model)
+    dists = np.linalg.norm(sim.data[con].eval_points, axis=1)
+    assert np.all(dists <= radius)
+    assert np.any(dists >= 0.9 * radius)
```

Comment:
I usually try to write the test before fixing the bug to have some automated method to reproduce the bug. Only in the second step I fix the bug. The test will also ensure that a bug is not reintroduced (it was overlooked once, so it might be overlooked again).

Of course it is impossible to test everything, but I'm not sure if it is a good idea to throw away tests that have already been written for some reason. As long as the tests do not hinder refactoring, having many tests doesn't seem to be a problem to me. Maybe we should think about a better organization instead of deleting tests?